wormhole-connect/src/config/types.ts

```diff
@@ -363,3 +367,54 @@ export type NttGroup = {
 };
 
 export type NttGroups = { [key: string]: NttGroup };
+
+// Token bridge foreign asset cache
+// Used in utils/sdkv2.ts
+
+type ForeignAssets<C extends ChainV2> = Record<string, TokenAddressV2<C>>;
+
+export class WrappedTokenAddressCache {
+  caches: Partial<Record<ChainV2, ForeignAssets<ChainV2>>>;
+
+  constructor(tokens: TokensConfig, converter: SDKConverter) {
+    this.caches = {};
+
+    // Pre-populate cache with values from built-in config
+    for (const key in tokens) {
+      const token = tokens[key];
+
+      for (const chain in token.foreignAssets) {
+        const foreignAsset = tokens[key].foreignAssets![chain];
+        const chainV2 = converter.toChainV2(chain as ChainName);
+        const addr = WormholeV2.parseAddress(chainV2, foreignAsset.address);
+        this.set(key, chainV2, addr);
+      }
+
+      // Cache it on its native chain too
+      if (token.tokenId) {
+        const nativeChainV2 = converter.toChainV2(token.nativeChain);
```

Comment:
Can there be a potential bug where `nativeChainV2` is returned null or undefined from the converter? You might wanna add a check below before caching it.

Reply:
it will throw if it gets invalid inputs. if that happens in this case it means the core config has bad data. I can catch here in case that happens.